### PR TITLE
execdetails: output write details in RuntimeStats

### PR DESCRIPTION
**Pull request by sticnarf**
Signed-off-by: Yilin Chen <sticnarf@gmail.com>

### What problem does this PR solve?

Issue Number: ref #34487

### What is changed and how it works?

Now, TiKV returns the detailed time breakdown of write requests. This PR outputs the returned results and output them in the runtime stats or slow query logs.

Now a slow log of a commit will contain more information, it will look like:

```
# Prewrite_time: 0.001431165 Get_latest_ts_time: 0.000137354 Slowest_commit_rpc_detail: {total:0.001s, region_id: 2, store: 127.0.0.1:20160, tikv_wall_time: 1.19ms, scan_detail: {get_snapshot_time: 21.3µs, rocksdb: {block: {}}}, write_detail: {store_batch_wait: 9.16µs, propose_send_wait: 0s, persist_log: {total: 830.9µs, write_leader_wait: 134ns, sync_log: 807.1µs, write_memtable: 1.79µs}, commit_log: 850.1µs, apply_batch_wait: 17.7µs, apply: {total:80.2µs, mutex_lock: 230ns, write_leader_wait: 0s, write_wal: 17.7µs, write_memtable: 14.4µs}}} Write_keys: 2 Write_size: 48 Prewrite_region: 1
```

A new panel showing the RPC layer latency is added 

image

### Check List

Tests 

- [x] Manual test (add detailed scripts or steps below)

Documentation

- [x] Changes Slow Log Format

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Extend slow log to show time breakdown in TiKV of write requests.
```

### CODE REVIEW

**Review by you06 — COMMENTED** [review]
I found the Greek letter "µ" is not well displayed when opening with less in a develop machine, not sure if this is a problem.
rest LGTM

**Comment by sticnarf** [comment]
> I found the Greek letter "µ" is not well displayed when opening with less in a develop machine

TiDB already uses "µs" in many places, I guess it's not a big problem...

**Review by you06 — APPROVED** [review]

File: metrics/grafana/tidb.json [hunk 1]
```diff
@@ -9311,6 +9311,117 @@
...
             "alignLevel": null
           }
+        },
+        {
+          "aliasColors": {},
+          "bars": false,
+          "dashLength": 10,
+          "dashes": false,
+          "datasource": "${DS_TEST-CLUSTER}",
+          "description": "Time spent on the RPC layer between TiDB and TiKV",
```
Comment (cfzjywxk) [inline metrics/grafana/tidb.json:9321]: Better to comment it includes tidb batch client time.

**Review by cfzjywxk — APPROVED** [review]

**Comment by cfzjywxk** [comment]
@sticnarf 
Another thing is that in the decoded plan details the execution details would also be used, for example
```
 └─StreamAgg_8               cop[tikv]        1               funcs:count(1)->Column#20                                             15              tikv_task:{proc max:1.29s, min:0s, p80:1.18s, p95:1.29s, iters:9778, tasks:17}, scan_detail: {total_process_keys: 10000000, total_process_keys_size: 890000000, total_keys: 10000017, rocksdb: {delete_skipped_count: 0, key_skipped_count: 10000000, block: {cache_hit_count: 25, read_count: 14349, read_byte: 283.9 MB}}}                 N/A             N/A
            └─IndexFullScan_19        cop[tikv]        10000000        table:T_CUSTOMER, index:PRIMARY(customer_id), keep order:false        10000000        tikv_task:{proc max:1.29s, min:0s, p80:1.18s, p95:1.29s, iters:9778, tasks:17}   
```

For the write details added for transaction commands including `prewrite`, `pessimistic_lock`, and `commit`, we may also need to re-design the executor execution details and make the enhanced information match the execution flow, for example how a pessimistic lock request could slow one `Next` operation.

**Pull request merged**